src/Membership/MembershipRequestHandler.php

```diff
@@ -48,38 +49,61 @@ public function __construct(
         $this->committeeManager = $committeeManager;
     }
 
-    public function handle(MembershipRequest $membershipRequest)
+    public function handle(MembershipRequest $membershipRequest): Adherent
     {
         $adherent = $this->adherentFactory->createFromMembershipRequest($membershipRequest);
+        $this->manager->persist($adherent);
+        $this->sendEmailValidation($adherent);
+
+        $this->dispatcher->dispatch(UserEvents::USER_CREATED, new UserEvent($adherent));
+
+        return $adherent;
+    }
+
+    public function sendEmailValidation(Adherent $adherent): void
+    {
         $token = AdherentActivationToken::generate($adherent);
 
-        $this->manager->persist($adherent);
         $this->manager->persist($token);
         $this->manager->flush();
 
         $activationUrl = $this->generateMembershipActivationUrl($adherent, $token);
         $this->mailer->sendMessage(AdherentAccountActivationMessage::createFromAdherent($adherent, $activationUrl));
+    }
+
+    public function register(MembershipRequest $membershipRequest): void
```

Comment:
This method name should be more specific like `registerAsAdherent` and the handle method could be rename as well to `registerAsUser`.